wiki/Beatmapping/Flow/en.md

```diff
@@ -0,0 +1,55 @@
+---
+tags:
+  - linear flow
+  - oval flow
+  - circular flow
+---
+
+# Flow
+
+**Flow** is a mapping concept that describes natural and intuitive movement patterns of a player's cursor during gameplay. Flow, or the lack thereof, is related to the [osu! game mode](/wiki/Game_mode/osu!) due to its two-dimensional playfield. By utilising the vast variety of [patterns](/wiki/Beatmap/Pattern) and slider shapes possible in osu! beatmaps, mappers are able to direct a player's cursor around the playfield in various ways, yielding different effects on the gameplay experience.
+
+<!-- TODO: this applies to osu!catch as well, the article needs expansion: https://github.com/ppy/osu-wiki/pull/7407#pullrequestreview-988657741 -->
+
+Flow is considered a **subjective** and often opinionated matter due to how different mappers and players design or perceive movement. Some aspects of flow may impact gameplay in different degrees depending on [input device](/wiki/Gameplay/Input_device), such as [graphics tablet](/wiki/Gameplay/Input_device/Graphics_tablet) or [mouse](/wiki/Gameplay/Input_device/Mouse), or the player's acquired skill or experience.
+
+The opposite of flow is called [antiflow](/wiki/Beatmapping/Antiflow), which conversely involves unintuitive or unnatural movement patterns.
+
+## Key components
+
+While movement preferences vary from one person to another, there are a few ideas that give the feeling and impression of flow on a majority of patterns or whole beatmaps.
+
+### Smooth movement
+
+Series of hit objects that require smooth movement, without sharp, abrupt turns, or fast changes of pace, can be considered comfortable to play. In such patterns, the player's cursor may follow a smooth curve throughout. These curves are usually characterised by circular or semicircular segments (so-called *oval* or *circular flow*) going either clockwise or anticlockwise. Beatmaps often switch the direction of the circular flow at regular intervals,[^circular-flow] where the transitions can often accompanied by "neutral" patterns to ease the transition to the next pattern.
+
+Linear movement may also be considered another form of smooth flow, albeit less common, since it does not enforce any specific cursor trajectory type.
+
+### Repetition
+
+Another form of natural movement is repetition. Patterns with similar repeated movements that only differ slightly, such as back-and-forth [jumps](/wiki/Beatmap/Pattern/Jump) or a series of slider pairs, are comfortable to play due to subsequent movements following already familiar paths, making them easy to predict. Repeated patterns may become more and more comfortable throughout a map for the same reason.
```

Comment:
i think this was trying to describe two things at the same time, with the "more comfortable with the map" thing. that said i'm not sure if the last sentence here is relevant to flow, although it may be true

Reply:
last sentence is valid